torchgeo/datasets/oscd.py

```diff
@@ -0,0 +1,242 @@
+        self.split = split
+        self.transforms = transforms
+        self.download = download
+        self.checksum = checksum
+
+        self._verify()
+
+        self.files = self._load_files()
+
+    def __getitem__(self, index: int) -> Dict[str, Tensor]:
+        """Return an index within the dataset.
+
+        Args:
+            index: index to return
+
+        Returns:
+            data and label at that index
+        """
+        files = self.files[index]
+        # TODO: implement choosing bands (right now assuming bands="all")
+        image1 = self._load_image(files["images1"])
+        image2 = self._load_image(files["images2"])
+        mask = self._load_target(str(files["mask"]))
+
+        image = torch.stack(tensors=[image1, image2], dim=0)
+        sample = {"image": image, "mask": mask}
+
+        if self.transforms is not None:
+            sample = self.transforms(sample)
+
+        return sample
+
+    def __len__(self) -> int:
+        """Return the number of data points in the dataset.
+
+        Returns:
+            length of the dataset
+        """
+        return len(self.files)
+
+    # TODO: this needs to be refactored
+    def _load_files(self) -> List[Dict[str, Union[str, Sequence[str]]]]:
+        regions = []
+        temp_split = "Test" if self.split == "test" else "Train"
+        labels_root = os.path.join(
+            self.root, f"Onera Satellite Change Detection dataset - {temp_split} Labels"
+        )
+        images_root = os.path.join(
+            self.root, "Onera Satellite Change Detection dataset - Images"
+        )
+        folders = glob.glob(os.path.join(labels_root, "*/"))
+        for folder in folders:
+            region = folder.split(os.sep)[-2]
+            mask = os.path.join(labels_root, region, "cm", "cm.png")
+            images1 = glob.glob(
+                os.path.join(images_root, region, "imgs_1_rect", "*.tif")
+            )
+            images2 = glob.glob(
+                os.path.join(images_root, region, "imgs_2_rect", "*.tif")
+            )
+            images1 = sorted(images1, key=sort_sentinel2_bands)
+            images2 = sorted(images2, key=sort_sentinel2_bands)
+            with open(os.path.join(images_root, region, "dates.txt")) as f:
+                dates = tuple(
+                    [line.split()[-1] for line in f.read().strip().splitlines()]
+                )
+
+            regions.append(
+                dict(
+                    region=region,
+                    images1=images1,
+                    images2=images2,
+                    mask=mask,
+                    dates=dates,
+                )
+            )
+
+        return regions
+
+    def _load_image(self, paths: Sequence[str]) -> Tensor:
+        """Load a single image.
+
+        Args:
+            path: path to the image
+
+        Returns:
+            the image
+        """
+        images = np.stack([rasterio.open(path).read() for path in paths], axis=0)
```

Comment:
use something like 
```
images = []
with rasterio.open(...) as f:
   ...
images = np.stack(...)
```
instead so that all the file handles are closed